src/Shared/Microsoft.AspNetCore.Razor.Test.Common/Language/IntegrationTests/RazorBaselineIntegrationTestBase.cs

```diff
@@ -156,6 +157,12 @@ protected void AssertLinePragmas(RazorCodeDocument codeDocument)
             var sourceMappings = csharpDocument.SourceMappings;
             foreach (var sourceMapping in sourceMappings)
             {
+                var content = codeDocument.Source.Text.GetSubText(new TextSpan(sourceMapping.OriginalSpan.AbsoluteIndex, sourceMapping.OriginalSpan.Length)).ToString();
+                if (string.IsNullOrWhiteSpace(content))
+                {
+                    continue;
```

Comment:
It seems there are sometimes `#line` pragmas emitted for whitespace. Presumably that doesn't happen always so we need to skip the validation - is there a reason it happens sometimes and not always?

Reply:
My thinking was thus: The existing design time code-gen sometimes doesn't emit line pragmas, but none of the existing tests happened to cover that scenario. I decided to update the test validation here (it now matches the runtime validation, just below this change) rather that updating design time to emit line pragmas for whitespace (which runtime does) largely because updating design time code gen seems quite redundant at this point.

Could be overruled by an actual compiler team member, or maybe runtime shouldn't emit line pragmas either (see my reply to your other comment though), but I don't really have an opinion about which is best.

Reply:
Not sure I understand 
>I decided to update the test validation here (it now matches the runtime validation, just below this change)

Because AFICT the only change is to skip the whitespace checks. 

My assumption is this wasn't failing because none of the previous tests hit it, but the test you added does now cause this validation to fail, unless you skip the whitespace? 

I'm fine with this change, but we also have an escape hatch to just not run verification on design time if it's preferred to do that.

Reply:
Yeah, that's what I meant. I updated the validation done by tests, to skip validating line pragmas on whitespace.